mobify/progressive-web-scaffold
# Code Split Vendor from Main

**JIRA**: [WEB-1236](https://mobify.atlassian.net/browse/WEB-1236)

## Changes
- Use `CommonsChunkPlugin` to split vendor scripts from app [as per this](https://github.com/mobify/lancome-progressive/pull/829)
- Manage asynchronous download for `vendors.js` and `main.js` [as per this](https://github.com/mobify/lancome-progressive/pull/829)

| Before | After |
| ------- | ----- |
| **243.63** (gzipped) ![screen shot 2017-03-06 at 4 10 59 pm](https://cloud.githubusercontent.com/assets/734535/23635949/84e67430-0287-11e7-9cab-f1ecd5ad7682.png) | **152kb** and **89.95** (gzipped)  ![screen shot 2017-03-06 at 4 09 46 pm](https://cloud.githubusercontent.com/assets/734535/23635941/64ac2c32-0287-11e7-998d-08455aa3af81.png) |

> For reference, Lancome's `main.js` and `vendor.js` are 178kb and 166kb (gzipped) respectively.


| Timeline Before | Timeline After |
| ---------------- | -------------- |
| ![screen shot 2017-03-06 at 4 34 02 pm](https://cloud.githubusercontent.com/assets/734535/23636546/cf741cfc-028a-11e7-9cf1-1626610bb427.png) | ![screen shot 2017-03-06 at 4 00 44 pm](https://cloud.githubusercontent.com/assets/734535/23635733/4642dfe4-0286-11e7-9f55-173ce3bc0aab.png) |


## How to test-drive this PR
- run `MOBIFY_ANALYZE=true npm run build`, ensure the `node_modules` scripts are moved to `vendor.js`
- [Preview Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=0)
- Ensure `vendor.js` and `main.js` load asynchronously and in parallel in Network tab
